### PR TITLE
CA-277850 move xenstore_watch to ezxenstore

### PR DESCRIPTION
**Pull request by lindig**
This moves the xenstore_watch module into ezxenstore to share it with
other components. Consequently, ezxenstore becomes a new dependency.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

File: lib/xs_watch.ml [hunk 1]
```diff
@@ -12,24 +12,11 @@
...
-let handle_of_uuid u =
-  let s = Uuidm.to_bytes u in
-  let h = Array.make 16 0 in
-  for i = 0 to 15 do
-    h.(i) <- int_of_char s.[i]
-  done;
-  h
+module XSW_Debug = Debug.Make(struct let name = "xenstore_watch" end)
+module XSW=Ez_xenstore_watch.Make(XSW_Debug)
+include XSW
```
Comment (edwintorok) [inline lib/xs_watch.ml]: Unless you want to use the XSW name, you could just do `include Ez_xenstore_watch.Make(XSW_Debug)`

File: xc/xenops_server_xen.ml [hunk 2]
```diff
@@ -3030,7 +3030,7 @@ module Actions = struct
   let domain_disappeared xc xs domid =
     let token = watch_token domid in
     List.iter (fun d ->
-        List.iter (Xenstore_watch.unwatch ~xs token) (watches_of_device d)
+        List.iter (Xs_watch.unwatch ~xs token) (watches_of_device d)
```
Comment (edwintorok) [inline xc/xenops_server_xen.ml]: Does the name `Xenstore_watch` clash with anything or you just wanted to make sure its using the new code by renaming the module?

Comment (lindig) [reply xc/xenops_server_xen.ml]: I could retain the name. This is a result from earlier exploration where there was a problem of having the same module name in ezxenstore and locally. Now the module in ezxenstore is Ez_xenstore_watch and there would be no problem.

So I'll use the `include` you suggested and keep the old name.

**Comment by lindig** [comment]
I had run a BST a while ago. The code that I removed is the code that is now in ezxenstore. This is also a single commit that would be easy to revert.

**Pull request merged**